File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.nio.Buffer;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Set;
import java.util.StringTokenizer;

//TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
// click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
public class Main {

    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        HashSet<Integer> set = new HashSet<>();

        // 횟수
        int N = Integer.parseInt(br.readLine());
        for(int i = 0; i < N; i++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            String command = st.nextToken();
            int x = 0;
            if(!command.equals("all") && !command.equals("empty")){
                x = Integer.parseInt(st.nextToken());
            }
            switch (command){
                case "add":
                    set.add(x);
                    break;
                case  "remove":
                    set.remove(x);
                    break;
                case "check":
                    if(set.contains(x)){
                        bw.write("1\n");
                    }else{
                        bw.write("0\n");
                    }
                    break;
                case "toggle":
                    if(set.contains(x)){
                        set.remove(x);
                    }else{
                        set.add(x);
                    }
                    break;
                case "all":
                    for (int j = 1; j <= 20; j++){
                        set.add(j);
                    }
                    break;
                case "empty":
                    set.clear();
            }
        }
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/11723. 집합/집합.py
import sys

m = int(sys.stdin.readline())
S = set()

for _ in range(m):
    temp = sys.stdin.readline().strip().split()
    
    if len(temp) == 1:
        if temp[0] == "all":
            S = set([i for i in range(1, 21)])
        else:
            S = set()
    
    else:
        f, x = temp[0], temp[1]
        x = int(x)

        if f == "add":
            S.add(x)
        elif f == "remove":
            S.discard(x)
        elif f == "check":
            print(1 if x in S else 0)
        elif f == "toggle":
            if x in S:
                S.discard(x)
            else:
                S.add(x)